### ISSUE TITLE
Support ES2015 default parameter for functions

### ISSUE THREAD

**Issue by buttonfly1000** [mozilla/rhino#678]
Hi.

I found such a js functon
```js
function f(b = 1) {return b;}
```
would yield an exception:
```
org.mozilla.javascript.EvaluatorException: missing ) after formal parameters
```

Would it be possible to support default parameter for functions?

Thanks!

**Comment by cbartolome** [comment]
Bootstrap 5 started using default parameters in its JavaScript code, so Selenium tests using the HtmlUnit driver all report the above error now, when using the un-minified code. The minified code produces this error:
```
com.gargoylesoftware.htmlunit.ScriptException: missing formal parameter (http://localhost:35193/webjars/bootstrap/5.0.1/js/bootstrap.bundle.min.js#6)
```

**Comment by p-bakker** [comment]
This case is blocking us from updating [test262](https://github.com/tc39/test262/tree/f94fc660cc3c59b1f2f9f122fc4d44b4434b935c) to their latest master, as the [harness code](https://github.com/tc39/test262/tree/f94fc660cc3c59b1f2f9f122fc4d44b4434b935c/harness) used by the tests in test262 make use of default parameters, which crashes the test suite.

**Comment by exabrial** [comment]
> Bootstrap 5 started using default parameters in its JavaScript code, so Selenium tests using the HtmlUnit driver all report the above error now, when using the un-minified code. The minified code produces this error:
> 
> ```
> com.gargoylesoftware.htmlunit.ScriptException: missing formal parameter (http://localhost:35193/webjars/bootstrap/5.0.1/js/bootstrap.bundle.min.js#6)
> ```

I seem to be getting the same error with bootstrap using the non-minified code. When you tested, what version of bootstrap were you using? I'm hoping they didn't switch to default parameters for the non-min version. Thank you!

**Comment by p-bakker** [comment]
Closed by #1640
